### PR TITLE
Add tests to identify compiler errors in gRPC tool

### PR DESCRIPTION
**Pull request by dilanSachi**
## Purpose
$subject
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests

Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/2395
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/2423

`testHelloWorldWithDependency` & `testHelloWorldWithReservedNames` are disabled due to the issues mentioned in [2423](https://github.com/ballerina-platform/ballerina-standard-library/issues/2395)

### CODE REVIEW

File: tooling-tests/src/test/java/io/ballerina/stdlib/grpc/tools/ToolingBidirectionalTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package io.ballerina.stdlib.grpc.tools;
```
Comment (MadhukaHarith92) [inline tooling-tests/src/test/java/io/ballerina/stdlib/grpc/tools/ToolingBidirectionalTest.java:1]: License header missing

File: tooling-tests/src/test/java/io/ballerina/stdlib/grpc/tools/ToolingClientTest.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
+package io.ballerina.stdlib.grpc.tools;
```
Comment (MadhukaHarith92) [inline tooling-tests/src/test/java/io/ballerina/stdlib/grpc/tools/ToolingClientTest.java:1]: License header missing

File: tooling-tests/src/test/java/io/ballerina/stdlib/grpc/tools/ToolingServerTest.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
+package io.ballerina.stdlib.grpc.tools;
```
Comment (MadhukaHarith92) [inline tooling-tests/src/test/java/io/ballerina/stdlib/grpc/tools/ToolingServerTest.java:1]: License header missing

File: tooling-tests/src/test/java/io/ballerina/stdlib/grpc/tools/ToolingTestUtils.java [hunk 4]
```diff
@@ -0,0 +1,216 @@
+package io.ballerina.stdlib.grpc.tools;
```
Comment (MadhukaHarith92) [inline tooling-tests/src/test/java/io/ballerina/stdlib/grpc/tools/ToolingTestUtils.java:1]: License header missing

File: tooling-tests/src/test/java/io/ballerina/stdlib/grpc/tools/ToolingUnaryTest.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
+package io.ballerina.stdlib.grpc.tools;
```
Comment (MadhukaHarith92) [inline tooling-tests/src/test/java/io/ballerina/stdlib/grpc/tools/ToolingUnaryTest.java:1]: License header missing

Comment (dilanSachi) [reply tooling-tests/src/test/java/io/ballerina/stdlib/grpc/tools/ToolingUnaryTest.java:19]: Added in https://github.com/ballerina-platform/module-ballerina-grpc/pull/560/commits/e1a62ce69607647067e3ab63774f16b29277f5e1

**Review by MadhukaHarith92 — DISMISSED** [review]
LGTM

**Review by daneshk — APPROVED** [review]

File: native/src/main/java/io/ballerina/stdlib/grpc/builder/BallerinaFileBuilder.java [hunk 6]
```diff
@@ -65,16 +66,19 @@
...
         setRootDescriptor(rootDescriptor);
         this.dependentDescriptors = dependentDescriptors;
+        streamClassMap = new HashMap<>();
     }
 
     public BallerinaFileBuilder(byte[] rootDescriptor, Set<byte[]> dependentDescriptors, String balOutPath) {
         setRootDescriptor(rootDescriptor);
         this.dependentDescriptors = dependentDescriptors;
         this.balOutPath = balOutPath;
+         streamClassMap = new HashMap<>();
```
Comment (BuddhiWathsala) [inline native/src/main/java/io/ballerina/stdlib/grpc/builder/BallerinaFileBuilder.java:81]: Seems not indented properly.

Comment (dilanSachi) [reply native/src/main/java/io/ballerina/stdlib/grpc/builder/BallerinaFileBuilder.java]: Fixed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/560/commits/3a103d34283678d8df0c60cd70bcc311836ac637

**Review by BuddhiWathsala — APPROVED** [review]

**Pull request merged**